### ISSUE TITLE
Authentication with O365 - Authentication does not work if O365 has multiple organisations

### ISSUE THREAD

**Issue by wucherpfennig** [zammad/zammad#2913]
### Infos:

* Used Zammad version: 3.2
* Installation method (source, package, ..): package
* Operating system: Debian 10.2
* Database + version: default
* Elasticsearch version: 7.5
* Browser + version: Chrome, latest

### Expected behavior:

* Authentication against O365 accounts which have access to multiple organisations / tenants

### Actual behavior:

![image](https://user-images.githubusercontent.com/3634653/73349118-ea2d9f80-428a-11ea-9d8d-78d748af6b63.png)

* Works only if in azure app registry the access permissions are set to "all"
* Analyzing azure concluded that in order to authenticate against a tenant you should also provide the tenant id

![image](https://user-images.githubusercontent.com/3634653/73348816-62e02c00-428a-11ea-9c79-ce259369547c.png)

**Assumtion**
A tentant ID should be provided by zammad (optional) in order to authenticate against the right organisation:

![image](https://user-images.githubusercontent.com/3634653/73349041-bd798800-428a-11ea-80e6-1d3f98be3b64.png)

### Steps to reproduce the behavior:

* Enable O365 auth

Yes I'm sure this is a bug and no feature request or a general question.

**Comment by MrGeneration** [comment]
In my opinion that's rather a feature request, as this functionality has been added by Microsoft within the last months.

What do you think @thorsteneckel ?

**Comment by thorsteneckel** [comment]
Yeah, it's a (valid) enhancement but not a bug. We need to check wether the gem we're using already supports this: 

https://rubygems.org/gems/omniauth-microsoft-office365
https://github.com/murbanski/omniauth-microsoft-office365/

On first glimpse it doesn't look like it.

**Comment by MrGeneration** [comment]
For the future, please post feature requests on the Community Forum.
However, as this does affect a broad mass of our users, I will keep this opened here.

ETA unknown.

**Comment by wucherpfennig** [comment]
@MrGeneration sure, will do! honestly this would not have come up if not for the issue #2912 ;-)

**Comment by MrGeneration** [comment]
No worries, glad you could identify this one as it has been appearing more often on Github and the Community than I did like. :P
